src/Views/Menu.java

```diff
@@ -187,34 +245,36 @@ private int addNewProduct() {
 
     /**
      * Prompt the user for a date input in the provided format.
+     *
      * @param start - boolean to tell which date to prompt the user for.
      * @return - user input.
      */
     private String dateInput(boolean start) {
         String dateType = start ? "start" : "end";
         System.out.println("Please enter a " + dateType + " date (YYYY-MM-DD)");
-        Scanner input = new Scanner(System.in);
-        return input.nextLine();
+        return scanner.nextLine();
     }
 
     /**
      * Prompt the user for an integer number of months since last purchase to consider the user for the email
      * promotion list.
+     *
      * @return - user input
      */
     private int intInput(String message) {
         System.out.print(message);
         Scanner input = new Scanner(System.in);
         String num = input.nextLine();
         while (!num.matches("^\\d+$")) {
-            num = input.nextLine();
+            num = scanner.nextLine();
```

Comment:
Ok I see what you meant about using the class Scanner instead of creating a new one.